vpr/src/place/analytical_placement/PartialPlacement.h

```diff
@@ -54,6 +54,12 @@ class PartialPlacement {
 
     double get_HPWL();
 
+    bool is_valid_partial_placement();
+
+    bool is_valid_node(size_t node_id);
+
+    void unicode_art();
```

Comment:
Some comments here would also be appreciated. Something to quickly explain what each method is doing.